### PR TITLE
Add add-ons view

### PR DESCRIPTION
**Pull request by magicmatatjahu**
**Description**

Changes proposed in this pull request:
- Add add-ons view:
  - update `luigi-config.js` for new view
  - write in `react` and `ts` new view for addons-configurations.

**Related issue(s)**
See also https://github.com/kyma-project/console/issues/751

### CODE REVIEW

File: add-ons/src/core/styled.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+import styled from 'styled-components';
```
Comment (aerfio) [inline add-ons/src/core/styled.ts]: this can be deleted if it's not used

File: add-ons/src/components/Modals/DeleteConfigurationModal/DeleteConfigurationModal.component.tsx [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    <Modal
+      title="Delete"
+      confirmText="Delete"
+      onConfirm={handleDelete}
+      modalOpeningComponent={openingComponentModal}
+      type="negative"
+      onShow={() => LuigiClient.uxManager().addBackdrop()}
+      onHide={() => LuigiClient.uxManager().removeBackdrop()}
+    >
+      {`Are you sure you want to delete configuration "${configurationName}"?`}
```
Comment (akucharska) [inline add-ons/src/components/Modals/DeleteConfigurationModal/DeleteConfigurationModal.component.tsx]: No hardcoded descriptions. Extract strings to separate file. Use variable here.

File: add-ons/src/components/Modals/DeleteUrlModal/DeleteUrlModal.component.tsx [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    <Modal
+      title="Delete"
+      confirmText="Delete"
+      onConfirm={handleDelete}
+      modalOpeningComponent={openingComponentModal}
+      type="negative"
+      onShow={() => LuigiClient.uxManager().addBackdrop()}
+      onHide={() => LuigiClient.uxManager().removeBackdrop()}
+    >
+      {`Are you sure you want to delete url of configuration "${configurationName}"?`}
```
Comment (akucharska) [inline add-ons/src/components/Modals/DeleteUrlModal/DeleteUrlModal.component.tsx]: No hardcoded descriptions. Extract strings to separate file. Use variable here.

File: add-ons/src/components/Modals/Modal.component.tsx [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+      </Button>
+      <Button
+        onClick={() => {
+          onSubmit && onSubmit();
+          setFalseShowModal();
+        }}
+        option="emphasized"
+        disabled={Boolean(onSubmitDisabled)}
+      >
+        {confirmText ? confirmText : 'Create'}
```
Comment (akucharska) [inline add-ons/src/components/Modals/Modal.component.tsx]: Move 'Create' to variables file

File: add-ons/src/components/Table/ActiveFilters.component.tsx [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+            {Object.keys(activeFilters.labels).map(key =>
+              activeFilters.labels[key].map(value => (
+                <StyledToken
+                  key={key}
+                  onClick={() => removeFilterLabel(key, value)}
+                >{`${key}=${value}`}</StyledToken>
+              )),
+            )}
+            <Button onClick={removeAllFiltersLabels} compact option="light">
+              Clear All
```
Comment (akucharska) [inline add-ons/src/components/Table/ActiveFilters.component.tsx]: No hardcoded descriptions. Extract strings to separate file. Use variable here.

File: add-ons/src/components/Table/Table.component.tsx [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+
+const TableComponent: React.FunctionComponent<TableComponentProps> = ({
+  configurationsExist,
+  children,
+}) => {
+  return (
+    <PanelWrapper>
+      <Panel>
+        <Panel.Header>
+          <Panel.Head title="Repository URLs" />
```
Comment (akucharska) [inline add-ons/src/components/Table/Table.component.tsx]: Take title from variable

File: add-ons/src/components/Toolbar/Toolbar.component.tsx [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+import { ActionBar, Button } from 'fundamental-react';
+
+import AddNewConfigurationModal from '../Modals/AddNewConfigurationModal/AddNewConfigurationModal.container';
+
+import { StyledActionBar } from './styled';
+
+const Toolbar: React.FunctionComponent = () => {
+  return (
+    <StyledActionBar>
+      <ActionBar.Header title="Add-Ons Configuration" />
```
Comment (akucharska) [inline add-ons/src/components/Toolbar/Toolbar.component.tsx]: Again :)

File: add-ons/src/constants.ts [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+
+export const BACKEND_MODULE_SERVICE_CATALOG = 'servicecatalogaddons';
+export const BACKEND_MODULE_SERVICE_CATALOG_DISPLAY_NAME =
+  'Service Catalog Addons';
+
+export const CONFIGURATION_VARIABLE = '{CONFIGURATION}';
+
+export const CONTENT_HEADERS: string[] = ['Name / URL', 'Labels', ''];
+
+export const NOTIFICATION = {
```
Comment (akucharska) [inline add-ons/src/constants.ts]: Very well :) Move to this file all other constants I marked

File: add-ons/src/services/Labels.service.ts [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+
+    const regex = /([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9]/;
+    const foundKey = key.match(regex);
+    const isKeyValid = Boolean(foundKey && foundKey[0] === key && key !== '');
+    const foundVal = value.match(regex);
+    const isValueValid = Boolean(
+      (foundVal && foundVal[0] === value) || value !== '',
+    );
+    if (!isKeyValid || !isValueValid) {
+      return `Invalid label ${key}=${value}! In a valid label, a key and value cannot be empty, a key/value consists of alphanumeric characters, '-', '_' or '.', and must start and end with an alphanumeric character.`;
```
Comment (akucharska) [inline add-ons/src/services/Labels.service.ts]: No hardcoded descriptions. Extract strings to separate file. Use variable here.

File: add-ons/src/services/Labels.service.ts [hunk 10]
```diff
@@ -0,0 +1,92 @@
...
+    }
+
+    const duplicateKeyExists: boolean = Boolean(
+      existingLabels
+        .map(l => l.split('=')[0])
+        .find((keyFromList: string) => keyFromList === key),
+    );
+    if (duplicateKeyExists) {
+      return `Invalid label ${key}=${value}! Keys cannot be reused!`;
```
Comment (akucharska) [inline add-ons/src/services/Labels.service.ts]: No hardcoded descriptions. Extract strings to separate file. Use variable here.

File: add-ons/src/components/Modals/AddNewConfigurationModal/AddNewConfigurationModal.component.tsx [hunk 11]
```diff
@@ -0,0 +1,180 @@
...
+        </FormItem>
+        <AddLabelButtonWrapper>
+          <Button
+            glyph="add"
+            onClick={addLabel}
+            option="light"
+            compact
+            disabled={Boolean(labelsField.error || !labelsField.value)}
+          >
+            Add Label
```
Comment (akucharska) [inline add-ons/src/components/Modals/AddNewConfigurationModal/AddNewConfigurationModal.component.tsx]: No hardcoded descriptions. Extract strings to separate file. Use variable here.

File: add-ons/src/components/Modals/AddNewConfigurationModal/AddNewConfigurationModal.component.tsx [hunk 12]
```diff
@@ -0,0 +1,180 @@
...
+          </Button>
+        </AddLabelButtonWrapper>
+        <FormItem key="url">
+          <FormLabel htmlFor="url" required>
+            Url
```
Comment (akucharska) [inline add-ons/src/components/Modals/AddNewConfigurationModal/AddNewConfigurationModal.component.tsx]: No hardcoded descriptions. Extract strings to separate file. Use variable here.

File: add-ons/src/components/Modals/AddNewConfigurationModal/AddNewConfigurationModal.component.tsx [hunk 13]
```diff
@@ -0,0 +1,180 @@
...
+            state={nameField.checkState()}
+            {...nameField.bind}
+          />
+          {nameField.error ? (
+            <FormMessage type="error">{nameField.error}</FormMessage>
+          ) : null}
+        </FormItem>
+        <FormItem key="labels">
+          <FormLabel htmlFor="labels">
+            Labels
```
Comment (akucharska) [inline add-ons/src/components/Modals/AddNewConfigurationModal/AddNewConfigurationModal.component.tsx]: No hardcoded descriptions. Extract strings to separate file. Use variable here.

File: add-ons/src/constants.ts [hunk 14]
```diff
@@ -0,0 +1,83 @@
...
+};
+
+export const HELP = {
+  URL_FIELD: '',
+  NAME_FIELD: '',
+  LABELS_FIELD: '',
+};
+
+export const PLACEHOLDERS = {
+  URL_FIELD: 'Insert repository URL.',
```
Comment (klaudiagrz) [inline add-ons/src/constants.ts]:
```suggestion
  URL_FIELD: 'Insert repository URL',
```

File: add-ons/src/constants.ts [hunk 15]
```diff
@@ -0,0 +1,83 @@
...
+  NAME_FIELD: 'Insert name.',
```
Comment (klaudiagrz) [inline add-ons/src/constants.ts]:
```suggestion
  NAME_FIELD: 'Insert name',
```

File: add-ons/src/constants.ts [hunk 16]
```diff
@@ -0,0 +1,83 @@
...
+  LABELS_FIELD: 'Insert labels (optionally).',
+  SEARCH_FIELD: 'Search configurations',
+};
+
+export const ERRORS = {
+  SERVER: 'Error with server. Please contact with admin of the cluster.',
```
Comment (klaudiagrz) [inline add-ons/src/constants.ts]:
```suggestion
  SERVER: 'Server error. Please contact the admin of the cluster.',
```

File: add-ons/src/constants.ts [hunk 17]
```diff
@@ -0,0 +1,83 @@
...
+  RESOURCES_NOT_FOUND: 'Resources not found.',
+  NOT_MATCHING_FILTERS: "Couldn't find resources with matching filters.",
+  NAME: {
+    ALREADY_EXISTS: 'Name already exists',
```
Comment (klaudiagrz) [inline add-ons/src/constants.ts]:
```suggestion
    ALREADY_EXISTS: 'This name already exists.',
```

File: add-ons/src/constants.ts [hunk 18]
```diff
@@ -0,0 +1,83 @@
...
+    REGEX:
+      "A name value must consists of alphanumeric characters, '-', '_' or '.', and must starts and ends with an alphanumeric character and contains maximum 63 chars.",
```
Comment (klaudiagrz) [inline add-ons/src/constants.ts]:
```suggestion
      "The name must consist of 3-63 characters, start and end with an alphanumeric character, and contain only alphanumeric characters, dashes, periods, and underscores.",
```

File: add-ons/src/constants.ts [hunk 19]
```diff
@@ -0,0 +1,83 @@
...
+    SHORT: 'Name must contains more than 3 chars.',
```
Comment (klaudiagrz) [inline add-ons/src/constants.ts]:
```suggestion
    SHORT: 'Name must contain more than 3 characters.',
```

File: add-ons/src/constants.ts [hunk 20]
```diff
@@ -0,0 +1,83 @@
...
+    LONG_NAME: 'Name must contains less than 64 chars.',
```
Comment (klaudiagrz) [inline add-ons/src/constants.ts]:
```suggestion
    LONG_NAME: 'Name must contain less than 64 characters.',
```

File: add-ons/src/constants.ts [hunk 21]
```diff
@@ -0,0 +1,83 @@
...
+  },
+  URL: {
+    ALREADY_EXISTS: 'Url already exists in config.',
```
Comment (klaudiagrz) [inline add-ons/src/constants.ts]:
```suggestion
    ALREADY_EXISTS: 'This URL already exists in the configuration.',
```

File: add-ons/src/constants.ts [hunk 22]
```diff
@@ -0,0 +1,83 @@
...
+    STARTS_WITH_HTTP: 'Url must starts with http(s) protocol.',
```
Comment (klaudiagrz) [inline add-ons/src/constants.ts]:
```suggestion
    STARTS_WITH_HTTP: 'URL must start with the http(s) protocol.',
```

File: add-ons/src/constants.ts [hunk 23]
```diff
@@ -0,0 +1,83 @@
...
+    YAML_EXTENSION: 'Url must have .yaml or .yml extension.',
```
Comment (klaudiagrz) [inline add-ons/src/constants.ts]:
```suggestion
    YAML_EXTENSION: 'URL must have .yaml or .yml extension.',
```

File: add-ons/src/constants.ts [hunk 24]
```diff
@@ -0,0 +1,83 @@
...
+    RESOURCE_NOT_EXISTS: 'Url not exists.',
```
Comment (klaudiagrz) [inline add-ons/src/constants.ts]:
```suggestion
    RESOURCE_NOT_EXISTS: 'URL doesn't exist.',
```

File: add-ons/src/constants.ts [hunk 25]
```diff
@@ -0,0 +1,83 @@
...
+  },
+};
+
+export const TOOLTIP_DATA_ERROR = {
+  type: 'error',
+  content: 'Fill out all mandatory fields',
```
Comment (klaudiagrz) [inline add-ons/src/constants.ts]:
```suggestion
  content: 'Fill out all mandatory fields.',
```

File: add-ons/dupa.yaml [hunk 26]
```diff
@@ -0,0 +1,17 @@
+apiVersion: "ui.kyma-project.io/v1alpha1"
```
Comment (klaudiagrz) [inline add-ons/dupa.yaml]: I guess you should rename this file :)

File: add-ons/public/index.html [hunk 27]
```diff
@@ -0,0 +1,21 @@
...
+    <meta
+      name="viewport"
+      content="width=device-width, initial-scale=1, shrink-to-fit=no"
+    />
+    <meta name="theme-color" content="#000000" />
+    <link rel="manifest" href="%PUBLIC_URL%/manifest.json" />
+    <title>Add-Ons UI</title>
+  </head>
+  <body>
+    <noscript>You need to enable JavaScript to run this app.</noscript>
```
Comment (klaudiagrz) [inline add-ons/public/index.html]:
```suggestion
    <noscript>Enable JavaScript to run this application.</noscript>
```

File: add-ons/src/components/Modals/DeleteConfigurationModal/DeleteConfigurationModal.component.tsx (hunk 2 above)
Comment (klaudiagrz) [inline add-ons/src/components/Modals/DeleteConfigurationModal/DeleteConfigurationModal.component.tsx]:
```suggestion
      {`Are you sure you want to delete the "${configurationName}" configuration?`}
```

File: add-ons/src/components/Modals/DeleteUrlModal/DeleteUrlModal.component.tsx (hunk 3 above)
Comment (klaudiagrz) [inline add-ons/src/components/Modals/DeleteUrlModal/DeleteUrlModal.component.tsx]:
```suggestion
      {`Are you sure you want to delete URL of the "${configurationName}" configuration?`}
```

File: add-ons/src/constants.ts [hunk 28]
```diff
@@ -0,0 +1,83 @@
...
+    CONTENT: `Configuration ${CONFIGURATION_VARIABLE} has been successfully updated`,
+  },
+  DELETE_CONFIGURATION: {
+    TITLE: 'Configuration removed',
+    CONTENT: `Configuration ${CONFIGURATION_VARIABLE} has been successfully removed`,
+  },
+};
+
+export const HELP = {
+  URL_FIELD: '',
```
Comment (klaudiagrz) [inline add-ons/src/constants.ts]:
```suggestion
  URL_FIELD: 'The URL must be unique for a given configuration and start with the http(s) protocol.',
```

File: add-ons/src/constants.ts [hunk 29]
```diff
@@ -0,0 +1,83 @@
...
+  NAME_FIELD: '',
```
Comment (klaudiagrz) [inline add-ons/src/constants.ts]:
```suggestion
  NAME_FIELD: 'The name must be unique, start and end with a lowercase letter, and contain only lowercase letters, numbers, dashes, periods, and underscores.',
```

File: add-ons/src/constants.ts [hunk 30]
```diff
@@ -0,0 +1,83 @@
...
+  LABELS_FIELD: '',
```
Comment (klaudiagrz) [inline add-ons/src/constants.ts]:
```suggestion
  LABELS_FIELD: 'The label must be a unique key=value pair for a given configuration, start and end with an alphanumeric character, and contain only alphanumeric characters, dashes, periods, and underscores.',
```

File: add-ons/src/services/nest.tsx [hunk 31]
```diff
@@ -0,0 +1,7 @@
+import React from 'react';
+
+export default (...components: any[]) => (props: any) =>
```
[the remark this answers is not in the document]
Comment (magicmatatjahu) [reply add-ons/src/services/nest.tsx]: Here we have only one function and only use in `index.ts`, so I think your proposition is unnecessary.

File: add-ons/src/components/Modals/AddNewConfigurationModal/AddNewConfigurationModal.container.tsx [hunk 32]
```diff
@@ -0,0 +1,165 @@
...
+import ConfigurationsService from '../../../services/Configurations.service';
+import LabelsService from '../../../services/Labels.service';
+import UrlsService from '../../../services/Urls.service';
+
+import Component from './AddNewConfigurationModal.component';
+
+import { ConfigurationLabels } from '../../../types';
+import { CONFIGURATION_NAME_PREFIX, ERRORS } from '../../../constants';
+
+interface Props {}
```
[the remark this answers is not in the document]
Comment (magicmatatjahu) [reply add-ons/src/components/Modals/AddNewConfigurationModal/AddNewConfigurationModal.container.tsx]: I left empty props for all containers (eg), because I had define some structure for all project.

File: add-ons/public/manifest.json [hunk 33]
```diff
@@ -0,0 +1,15 @@
+{
```
[the remark this answers is not in the document]
Comment (magicmatatjahu) [reply add-ons/public/manifest.json]: I forgot about this. Thanks :)

File: add-ons/src/services/nest.tsx (hunk 31 above)
[the remark this answers is not in the document]
Comment (aerfio) [reply add-ons/src/services/nest.tsx]: Ok, fair enough 👍

File: add-ons/tslint.json [hunk 34]
```diff
@@ -0,0 +1,42 @@
+{
+  "defaultSeverity": "error",
+  "extends": ["tslint:recommended"],
```
Comment (aerfio) [inline add-ons/tslint.json]: Why did you remove `"tslint-react", "tslint-config-prettier"`? Add them back in please

File: add-ons/Makefile [hunk 35]
```diff
@@ -0,0 +1,27 @@
...
+resolve:
+	npm install
+
+build:
+	../scripts/prepare-react-components.sh -d react-components
+	npm run build
+
+validate:
+	npm run lint-check
+
```
Comment (aerfio) [inline add-ons/Makefile]: Add `npm run conflict-check` here too

File: add-ons/tslint.json [hunk 36]
```diff
@@ -0,0 +1,42 @@
...
+    "member-access": false,
+    "indent": false,
+    "ordered-imports": [false],
+    "no-implicit-dependencies": false,
+    "no-unnecessary-initializer": false,
+    "unified-signatures": false,
+    "callable-types": false,
+    "max-line-length": [150],
+    "member-ordering": [false],
+    "curly": false,
```
Comment (aerfio) [inline add-ons/tslint.json]: `true`, or at least `[true, "ignore-same-line"]`, and I won't change my mind here

File: add-ons/tslint.json (hunk 34 above)
Comment (aerfio) [reply add-ons/tslint.json]: You have a lot of conflicts with our `.prettier.rc` because you deleted `"tslint-react", "tslint-config-prettier"` and added them yourself. And I see you added a lot of rules that are kinda good, but make it look too strict, like `eofline`, `quotemark`, `max-line-length` etc.

File: add-ons/tslint.json [hunk 37]
```diff
@@ -0,0 +1,42 @@
...
+    "variable-name": [false],
+    "one-line": [false],
+    "one-variable-per-declaration": [false],
+    "no-return-await": true,
+    "prefer-readonly": true,
+    "no-string-literal": false,
+    "interface-over-type-literal": false,
+    "no-trailing-whitespace": false,
+    "no-shadowed-variable": false,
+    "semicolon": false
```
Comment (aerfio) [inline add-ons/tslint.json]: That rule is not even enforced in codebase, look through it, there's `;` at the end of every statement

It would be best if you just copied `tslint.json` from `components/react-odata` as I have spent some time to pick non-conflicting rules, and it would be consistent with what we already have, we could even extract `tslint.json` to root, alongside `.prettierrc`

File: add-ons/package.json [hunk 38]
```diff
@@ -0,0 +1,87 @@
...
+    "click-outside-hook": "^1.1.0",
+    "constate": "^1.0.0",
+    "fiori-fundamentals": "^1.5.2",
+    "fundamental-react": "^0.5.0",
+    "graphql": "^14.2.1",
+    "graphql-tag": "^2.10.0",
+    "react": "^16.8.6",
+    "react-apollo-hooks": "^0.4.5",
+    "react-dom": "^16.8.6",
+    "styled-components": "^3.4.10",
```
[the remark this answers is not in the document]
Comment (magicmatatjahu) [reply add-ons/package.json]: Yes, because in `components/react` we still use version `~3.0.0` and `^=4.0.0` has a breaking change with API, that we used.

File: add-ons/src/components/Atoms/InlineHelp.tsx [hunk 39]
```diff
@@ -0,0 +1,15 @@
...
+
+const InlineHelp: React.FunctionComponent<Props> = ({ text }) => (
+  <span className="fd-inline-help fd-has-float-right">
+    <span className="fd-inline-help__content fd-inline-help__content--bottom-left">
+      {text}
+    </span>
+  </span>
+);
+
+export default InlineHelp;
```
[the remark this answers is not in the document]
Comment (magicmatatjahu) [reply add-ons/src/components/Atoms/InlineHelp.tsx]: Yes, Concur have it XD But I take it from playground `fundamental-react`. https://sap.github.io/fundamental-react/forms#_input-_help-_elements

File: add-ons/src/components/Toolbar/Toolbar.component.tsx [hunk 40]
```diff
@@ -0,0 +1,21 @@
+import React from 'react';
+import { ActionBar, Button } from 'fundamental-react';
```
[the remark this answers is not in the document]
Comment (magicmatatjahu) [reply add-ons/src/components/Toolbar/Toolbar.component.tsx]: Yes, we have but we merge `0.5.0` version 1 week after I started task.

File: add-ons/src/components/Modals/AddUrlModal/AddUrlModal.container.tsx [hunk 41]
```diff
@@ -0,0 +1,112 @@
...
+
+    const existingUrls = [...getUrlsFromConfigByName(getConfigName()), ...urls];
+    return validateUrl(url, existingUrls);
+  };
+  const handleEnterDownOnUrlField = (event: any) => {
+    if (event.key === 'Enter' && !urlField.error) {
+      addUrl();
+    }
+  };
+  const urlField = useInput('', validateUrlField);
```
[the remark this answers is not in the document]
Comment (magicmatatjahu) [reply add-ons/src/components/Modals/AddUrlModal/AddUrlModal.container.tsx]: You must see 2 things:
- I pass `validateUrlField ` function as second parameter to this hook, so I must define it before definition urlField - this is not a problem in JS, but probably is in TS...
- because JS has a [hoisting mechanism](https://developer.mozilla.org/pl/docs/Glossary/Hoisting) so this is not a problem for language (but in TS problem is declaration of function) :)

File: add-ons/src/components/Modals/AddNewConfigurationModal/AddNewConfigurationModal.container.tsx [hunk 42]
```diff
@@ -0,0 +1,165 @@
...
+
+import { useInput } from '../../../services/Forms';
+import {
+  MutationsService,
+  ConfigurationsService,
+  LabelsService,
+  UrlsService,
+} from '../../../services';
+
+import Component from './AddNewConfigurationModal.component';
```
[the remark this answers is not in the document]
Comment (magicmatatjahu) [reply add-ons/src/components/Modals/AddNewConfigurationModal/AddNewConfigurationModal.container.tsx]: I miss about it, thanks.

File: add-ons/src/components/Table/Search/Search.container.tsx [hunk 43]
```diff
@@ -0,0 +1,40 @@
...
+  const { configurationsExist } = useContext(ConfigurationsService);
+  const {
+    setSearchFilter,
+    activeFilters: { search },
+  } = useContext(FiltersService);
+
+  const searchField = useInput('');
+  const [showSearchIcon, setShowSearchIcon] = useState<boolean>(true);
+
+  const reference = useClickOutside(() => {
```
[the remark this answers is not in the document]
Comment (magicmatatjahu) [reply add-ons/src/components/Table/Search/Search.container.tsx]: 👌

File: add-ons/src/components/Table/Table.component.tsx [hunk 44]
```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+import { Panel, Button, Popover } from 'fundamental-react';
```
[the remark this answers is not in the document]
Comment (magicmatatjahu) [reply add-ons/src/components/Table/Table.component.tsx]: Yes, we have, but your merged `0.5.0` version 1 week after I started task.

File: add-ons/src/constants.ts [hunk 45]
```diff
@@ -0,0 +1,126 @@
...
+    REGEX:
+      'The name must consist of 4-63 characters, start and end with an alphanumeric character, and contain only alphanumeric characters, dashes, periods, and underscores.',
+    SHORT: 'Name must contain more than 3 characters.',
+    LONG_NAME: 'Name must contain less than 64 characters.',
+  },
+  URL: {
+    ALREADY_EXISTS: 'This URL already exists in the configuration.',
+    STARTS_WITH_HTTP: {
+      DEVELOPMENT_MODE: 'URL must start with the http(s) protocol.',
+      NOT_DEVELOPMENT_MODE: 'URL must start with the https protocol.',
```
[the remark this answers is not in the document]
Comment (magicmatatjahu) [reply add-ons/src/constants.ts]: In development you https or http, but in production only https.

File: add-ons/src/constants.ts [hunk 46]
```diff
@@ -0,0 +1,126 @@
...
+  ADD_URL_MODAL_TITLE: 'Add URL',
+  ADD_NEW_CONFIGURATION_BUTTON_TITLE: 'Add New Configuration',
+  ADD_NEW_CONFIGURATION_MODAL_TITLE: 'New Configuration',
+  DELETE_MODAL_TITLE: 'Delete',
+};
+
+const HELP_URL_FIELD_DEVELOPMENT_MODE =
+  'The URL must be unique for a given configuration and start with the http(s) protocol.';
+const HELP_URL_FIELD_NOT_DEVELOPMENT_MODE =
+  'The URL must be unique for a given configuration and start with the https protocol.';
```
[the remark this answers is not in the document]
Comment (magicmatatjahu) [reply add-ons/src/constants.ts]: In development you https or http, but in production only https.

File: add-ons/src/constants.ts [hunk 47]
```diff
@@ -0,0 +1,126 @@
...
+
+export const PLACEHOLDERS = {
+  URL_FIELD: 'Insert repository URL',
+  NAME_FIELD: 'Insert name',
+  LABELS_FIELD: 'Insert labels (optional)',
+  SEARCH_FIELD: 'Search configurations',
+};
+
+export const ERRORS = {
+  SERVER: 'Server error. Please contact the admin of the cluster',
```
Comment (klaudiagrz) [inline add-ons/src/constants.ts]:
```suggestion
  SERVER: 'Server error. Please contact the admin of the cluster.',
```

File: add-ons/src/constants.ts [hunk 48]
```diff
@@ -0,0 +1,126 @@
...
+    ALREADY_EXISTS: 'This URL already exists in the configuration.',
+    STARTS_WITH_HTTP: {
+      DEVELOPMENT_MODE: 'URL must start with the http(s) protocol.',
+      NOT_DEVELOPMENT_MODE: 'URL must start with the https protocol.',
+    },
+    YAML_EXTENSION: 'URL must have .yaml or .yml extension.',
+    RESOURCE_NOT_EXISTS: "URL doesn't exist.",
+  },
+  LABEL: {
+    INVALID_LABEL: `Invalid label ${LABEL_VARIABLE}! A key and value should be separated by a '='`,
```
Comment (klaudiagrz) [inline add-ons/src/constants.ts]:
```suggestion
    INVALID_LABEL: `Invalid ${LABEL_VARIABLE} label. Key and value should be separated by the equality sign.`,
```

File: add-ons/src/constants.ts [hunk 49]
```diff
@@ -0,0 +1,126 @@
...
+    KEY_OR_VALUE_INVALID: `Invalid label ${KEY_VARIABLE}=${VALUE_VARIABLE}! In a valid label, a key and value cannot be empty, a key/value consists of alphanumeric characters, '-', '_' or '.', and must start and end with an alphanumeric character.`,
```
Comment (klaudiagrz) [inline add-ons/src/constants.ts]:
```suggestion
    KEY_OR_VALUE_INVALID: `Invalid ${KEY_VARIABLE}=${VALUE_VARIABLE} label. In a valid label, key and value cannot be empty, must start and end with an alphanumeric character, and contain only alphanumeric characters, dashes, periods, and underscores.`,
```

File: add-ons/src/constants.ts [hunk 50]
```diff
@@ -0,0 +1,126 @@
...
+    DUPLICATE_KEYS_EXISTS: `Invalid label ${KEY_VARIABLE}=${VALUE_VARIABLE}! Keys cannot be reused!`,
```
Comment (klaudiagrz) [inline add-ons/src/constants.ts]:
```suggestion
    DUPLICATE_KEYS_EXISTS: `Invalid ${KEY_VARIABLE}=${VALUE_VARIABLE} label. Keys cannot be reused.`,
```
